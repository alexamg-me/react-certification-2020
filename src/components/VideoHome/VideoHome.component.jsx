import React, { useContext } from 'react';
import { Wrapper } from './VideoHome.styled';
import { GlobalContext } from '../../context/GlobalContext';
import { useSearch } from '../../hooks/useSearch';
import VideoList from '../VideoList';

function VideoHome() {
  const globalContext = useContext(GlobalContext);
  const cQuery = globalContext.query === '' ? 'wizeline' : globalContext.query;
  // console.log('VideoHome:cQuery', cQuery);
  const { data, error, loading } = useSearch(cQuery);
  // console.log('VideoDetail:demoData', data, error, loading);

  if (loading) return <h1>Loading...</h1>;
  if (error) return <h1>Hi Error {JSON.stringify(error)}</h1>;

  return <Wrapper>{data && <VideoList items={data.items} />}</Wrapper>;
}

export default VideoHome;
