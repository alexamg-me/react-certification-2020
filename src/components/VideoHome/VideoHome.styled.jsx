import styled from 'styled-components';

const Wrapper = styled.div`
  display: inline-block;
`;
const VideoList = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
`;

export { Wrapper, VideoList };
