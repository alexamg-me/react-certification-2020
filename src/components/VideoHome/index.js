export { default } from './VideoHome.component';
