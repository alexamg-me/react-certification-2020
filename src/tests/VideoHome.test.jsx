import React from 'react';
import TestRenderer from 'react-test-renderer';
import VideoHome from '../components/VideoHome/VideoHome.component';

describe(`Component: VideoHome`, () => {
  test(`VideoHome renders with default props`, () => {
    const component = TestRenderer.create(<VideoHome />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
