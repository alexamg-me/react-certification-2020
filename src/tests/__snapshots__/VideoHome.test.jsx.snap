// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: VideoHome VideoHome renders with default props 1`] = `
<div
  className="sc-hKgILt bBirOn"
>
  <div
    className="sc-eCssSg fRcATq"
  >
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Video Tour | Welcome to Wizeline Guadalajara
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/HYyRZiwBWc8/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline Guadalajara | Bringing Silicon Valley to Mexico
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/Po3VwR_NNGk/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline hace sentir a empleados como en casa
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/7PtYNO6g7eI/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        We Are Wizeline
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/YuW0CE_8i1I/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline Thrives in Mexico City
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/CHzlSGRvWPs/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/cjO2fJy8asM/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        A Day in the Life of an Engineering Manager at Wizeline
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/zClI9OjgKXM/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline Technical Writing Academy | Featuring Eduardo Ocejo
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/8bz9R61oY5o/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Silicon Valley en México
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/7dJFraOqcoQ/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Why Wizeline? featuring Juan Pablo Villa in Mexico City
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/w-Qwc_XJrWc/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline's 2020 Year in Review
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/rjir_cHTl5w/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Why Wizeline? featuring Hugo Lopez in Mexico City
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/DcFK1x3NHGY/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Why Wizeline? (We're Hiring in Mexico!)
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/3BzYWAqZgFw/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Why Wizeline? featuring Oswaldo Herrera in Mexico City
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/3KVFmT-Tp2w/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Caso de Éxito AWS: Wizeline [Spanish]
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/aKuPmY2m1Ro/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline's 2019 Year in Review
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/24sTHUyWhRM/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        [1 of 2] Wizeline CEO shares career lessons from Google
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/IxGc1gSqB3A/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline Data Engineering featuring Tania Reyes
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/NP1gAnbeNno/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Wizeline Querétaro | Mexico's New Knowledge Economy (We're hiring!)
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/F6Krwu6lUc8/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Miriam Godinez | Women in Leadership at Wizeline
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/RFq7gfvhtCk/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Welcome Back to Wizeline Vietnam | Extended Version
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/E1Vq_A3WKK8/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        How does Wizeline work?
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/ZmkslANDz0Q/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        12 Wishes from Wizeline | Happy Holidays 2019
      </p>
    </div>
    <div
      className="sc-bdfBwQ ivAyjI"
    >
      <img
        className="sc-dlfnbm kGZZzt"
        src="https://i.ytimg.com/vi/Nss3EmTDD3s/hqdefault.jpg"
      />
      <p
        className="sc-gsTCUz kwCLfT"
      >
        Why Wizeline?
      </p>
    </div>
  </div>
</div>
`;
